homeassistant/components/netatmo/sensor.py

```diff
@@ -132,7 +132,10 @@ def _retry(_data):
 
         import pyatmo
         for data_class in [pyatmo.WeatherStationData, pyatmo.HomeCoachData]:
-            data = NetatmoData(auth, data_class, config.get(CONF_STATION))
+            try:
+                data = NetatmoData(auth, data_class, config.get(CONF_STATION))
+            except pyatmo.NoDevice:
+                continue
```

Comment:
Should we log a warning here?

Reply:
I don't think so, because it most likely just means that the user doesn't have one of the two device types (home coach or weather station). But we can log a warning stating just that, like "No Home Coach devices found" if you like that better.